### PR TITLE
Improve TLDQueryIterator's EventDataQueryFilter

### PR DESCRIPTION
**Pull request by apmoriarty**
- refactored logic to only parse keys that are part of the root document
- removed DatawaveKey as parse mechanism
- added FiKeyUtil and test for parsing logical components from field index keys

### CODE REVIEW

**Review by FineAndDandy — DISMISSED** [review]

File: warehouse/query-core/src/test/java/datawave/core/iterators/key/util/FiKeyUtilTest.java [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+    @Test(expected = IllegalArgumentException.class)
+    public void testParseNoDatatype() {
+        Key key = new Key("row", "fi\0FIELD", "value\0uid");
+        assertEquals("datatype", FiKeyUtil.getDatatypeString(key));
+    }
+    
+    @Test(expected = IllegalArgumentException.class)
+    public void testParseNoUid() {
+        Key key = new Key("row", "fi\0FIELD", "value\0datatype");
+        assertEquals("uid", FiKeyUtil.getDatatypeString(key));
```
Comment (magrable) [inline warehouse/query-core/src/test/java/datawave/core/iterators/key/util/FiKeyUtilTest.java]: Is it worth noting that getUidString() will not throw an exception, but incorrectly parse the datatype as the uid in this case?

File: warehouse/query-core/src/main/java/datawave/query/tld/TLDQueryIterator.java [hunk 2]
```diff
@@ -128,13 +128,16 @@ public boolean peek(@Nullable Map.Entry<Key,String> var1) {
...
+             *            a field index key
+             * @return true if the key should be kept for evaluation
              */
             @Override
             public boolean keep(Key k) {
                 boolean root = TLDEventDataFilter.isRootPointer(k);
-                DatawaveKey datawaveKey = new DatawaveKey(k);
-                return (root && getIndexOnlyFields().contains(datawaveKey.getFieldName())) || !root;
+                if (root) {
+                    return getIndexOnlyFields().contains(FiKeyUtil.getFieldString(k));
```
Comment (magrable) [inline warehouse/query-core/src/main/java/datawave/query/tld/TLDQueryIterator.java:138]: Maybe this is obvious.. Is the parameter guaranteed to be a field index key?

Comment (apmoriarty) [reply warehouse/query-core/src/main/java/datawave/query/tld/TLDQueryIterator.java:138]: In this case you are guaranteed to only have field index keys.

File: warehouse/query-core/src/test/java/datawave/core/iterators/key/util/FiKeyUtilTest.java (hunk 1 above)
Comment (apmoriarty) [reply warehouse/query-core/src/test/java/datawave/core/iterators/key/util/FiKeyUtilTest.java]: Good catch -- this test is wrong. I'll push up a fix shortly.

**Review by magrable — APPROVED** [review]

**Review by FineAndDandy — APPROVED** [review]

**Pull request merged**